Update end to end tests

f2prateek opened the pull request:
Use the testing harness that uses the new schema webhook.

f2prateek commented on Makefile:
```diff
@@ -9,9 +9,14 @@ dependencies:
-check:
+check: install  # Installation required for testing binary
 	bundle exec rake
+	sh .buildscript/e2e.sh
```
It may be to nice run this e2e test only on CI, since it'll fail locally if you're using a non-Linux machine.

rohitpaulk replied:
Is it okay to rely on the `RUN_E2E_TESTS` env var not being set? The script will terminate early if so.

rohitpaulk commented:
One note - now that we've removed tests with the e2e tag, we should remove [this line](https://github.com/segmentio/analytics-ruby/blob/244e1adf40e1936c7bc475c17400fc2b9e619009/Rakefile#L7) too

f2prateek commented:
@rohitpaulk I removed this line as you suggested https://github.com/segmentio/analytics-ruby/blob/244e1adf40e1936c7bc475c17400fc2b9e619009/Rakefile#L7. I also updated the tester to use env vars instead of cli args. Looks like I may have broken something - mind taking another look when you get the chance?

f2prateek commented:
weird, went back to flags and it looks like something isn't quite working still :(

rohitpaulk commented:
Hmm, weird indeed... Will take a look today/tomorrow.

f2prateek commented:
thanks!

f2prateek commented:
I was able to fix the issue!
